scripts/jenkins/groovy/insideDocker.groovy

```diff
@@ -19,13 +19,29 @@ def call(customEnv, image, registry, buildConfig, timeoutValue, timeoutUnit, cus
     timeout(time: timeoutValue, unit: timeoutUnit) {
       docker.withRegistry("https://${registry}") {
         withCredentials([file(credentialsId: 'c096a055-bb45-4dac-ba5e-10e6e470f37e', variable: 'JUNIT_CORE_SITE_PATH'), [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AWS S3 Credentials', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
-          docker.image(image).inside("--init --add-host=nexus:172.17.0.53 --dns ${DEFAULT_DNS} -e AWS_ACCESS_KEY_ID=\${AWS_ACCESS_KEY_ID} -e AWS_SECRET_ACCESS_KEY=\${AWS_SECRET_ACCESS_KEY} -v /home/0xdiag:/home/0xdiag -v /home/jenkins/repos:/home/jenkins/repos ${customArgs}") {
+          if(dind) {
+            dockerGroupName = "docker"
+
+            dockerGroupId = sh(script: "awk -F: '/$dockerGroupName/ {print \$3}' /etc/group", returnStdout: true).trim()
+            if (dockerGroupId == "") {
+              error "cannot find gid of $dockerGroupName"
+            }
+            docker.image(image).inside("--init --group-add ${dockerGroupId} --add-host=nexus:172.17.0.53 --dns ${DEFAULT_DNS} -e AWS_ACCESS_KEY_ID=\${AWS_ACCESS_KEY_ID} -e AWS_SECRET_ACCESS_KEY=\${AWS_SECRET_ACCESS_KEY} -v /home/0xdiag:/home/0xdiag -v /home/jenkins/repos:/home/jenkins/repos ${customArgs}") {
```

Comment:
what about appending "--group-add ${dockerGroupId}" to customArgs (or putting into a new var that by default empty string), this way we do not have to duplicate the wholte docker.inside code

Reply:
We don't know the docker GID when defining the test stage. But we don't have to duplicate it. `dockerGroupId` can be added to the args every time, but will be empty by default. It should be filled-in only when `dind == true`. Does that make sense?

Reply:
Yes, makes sense. I'll do it exactly like that.